idelib/dataset.py

```diff
@@ -2500,6 +2502,34 @@ def getValueAt(self, at, outOfRange=False, display=False):
             v1 + (percent * (v2-v1))
             for v1, v2 in zip(startEvt[1:], endEvt[1:])
         )
+
+
+    def getMean(self, startTime=None, endTime=None, display=False, iterator=iter):
+        """ Get the mean value of all events, optionally within a specified
+            time range. For Channels, returns the minimum among all
+            Subchannels.
+
+            :keyword startTime: The starting time. Defaults to the start.
+            :keyword endTime: The ending time. Defaults to the end.
+            :keyword display: If `True`, the final 'display' transform (e.g.
+                unit conversion) will be applied to the results.
+            :return: The event with the minimum value.
+        """
+        if not self.hasMinMeanMax:
+            self._computeMinMeanMax()
+
+        if startTime is None and endTime is None:
+            if self._mean is not None:
+                return self._mean
+
+        means = self.arrayMinMeanMax(startTime, endTime, times=False,
+                                     display=display, iterator=iterator)[1]
+        mean = np.sum([d.numSamples*mean for d, mean in zip(self._data, means[0])])/np.sum(d.numSamples for d in self._data)
```

Comment:
actually, I just realized that since this is a new method, I probably just copied the docstring from `getmin` and forgot to update it.

Reply:
oh I didn't see the `times=False` bit. so that would change my most recent snippet to use `means` instead of `means[1:]`, making it look like this?
```python
        means = self.arrayMinMeanMax(startTime, endTime, times=False,
                                     display=display, iterator=iterator)[1]
        mean = np.amin(np.average(means, weights=[d.numSamples for d in self._data], axis=-1))
```

Reply:
yeah, that looks good, I just made a change to the code and I'll be pushing it.  Only difference is that the docstring was wrong, for Channels it's the mean of all the values in the channel, not the min of the means of the Subchannels.

Reply:
Ok, double checked, I think something got lost translating this to the new method.  I don't 100% know why, but the math checks out now.

Reply:
sweet! looks good now